core/src/main/java/com/amazon/opendistroforelasticsearch/sql/expression/operator/predicate/UnaryPredicateOperator.java

```diff
@@ -149,4 +163,8 @@ public static ExprValue exprNullIf(ExprValue v1, ExprValue v2) {
     return v1.equals(v2) ? LITERAL_NULL : v1;
   }
 
+  public static ExprValue exprIf(ExprValue v1, ExprValue v2, ExprValue v3) {
+    return v1.value().equals(LITERAL_TRUE.value()) ? v2 : v3;
```

Comment:
For `v1`, I think we need to do null/missing check? Please check MySQL behavior and if ours is the same.

And `ExprBooleanValue` already implements `equals()` so just do `LITERAL_TRUE.equals(v1)`?

Reply:
Thanks @dai-chen, you are correct, we need to do null/missing check. let me update both.

https://www.w3resource.com/mysql/control-flow-functions/if-function.php
IF() function
MySQL IF() takes three expressions and if the first expression is true, not zero and not NULL, it returns the second expression. Otherwise, it returns the third expression.

Depending on the context in which it is used, it returns either numeric or string value.